repair/repair.cc

```diff
@@ -1304,8 +1304,9 @@ future<> repair::user_requested_repair_task_impl::run() {
             }
             return make_ready_future<>();
         }).get();
-    }).handle_exception([id] (std::exception_ptr ep) {
+    }).handle_exception([id, &rs] (std::exception_ptr ep) {
         rlogger.warn("repair[{}]: user-requested repair failed: {}", id.uuid(), ep);
+        rs.get_repair_module().check_in_shutdown();
```

Comment:
Why is this needed?

Reply:
`check_in_shutdown` calls `check` on abort source and so if abort was requested it throws `abort_requested_exception` instead of exceptions from all shards wrapped.

Reply:
Line 1275 is reached only when repair was aborted with force_terminate_all_repair_sessions api

Reply:
I think this is worth a comment, also at the other similar place in this patch. It is suprising that we are discarding the current exception for a new one, so we need to commend to explain why this is ok.